src/Compilers/Core/Portable/PEWriter/PeWriter.cs

```diff
@@ -66,7 +66,7 @@ internal static class PeWriter
 
             var ilBuilder = new BlobBuilder(32 * 1024);
             var mappedFieldDataBuilder = new BlobBuilder();
```

Comment:
Why aren't these BlobBuilders pooled as well?

Reply:
The `ilBuilder` instance is not pooled because it is significantly larger than the instances that we have pools for. I considered creating a separate pool for `ilBuilder` but it's complicated. I wanted to get past this change, dig deeper into the profiles and if this proves to be a significant source of allocations then I'll come back to it. 

The `mappedFieldDataBuilder` wasn't shown to be a significant source of allocations hence I hadn't done the work yet to establish the ownership. Unfortunately unlike most other pooling decisions this isn't a simple "it's that type, use the pool" decision. Instead have to sit down and map out which `BlobBuilder` actually owns the instance.